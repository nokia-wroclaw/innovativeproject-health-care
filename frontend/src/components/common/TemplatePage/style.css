.template-page-container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.template-page-content {
  flex: 1;
}
