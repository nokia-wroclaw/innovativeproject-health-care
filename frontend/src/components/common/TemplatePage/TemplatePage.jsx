import React from "react";
import Header from "../Header";
import Footer from "../Footer";
import LoginModal from "../LoginModal";
import "./style.css";

const TemplatePage = ({ children }) => {
  return (
    <div className="template-page-container">
      <LoginModal />
      <Header />
      <div className="template-page-content">{children}</div>
      <Footer />
    </div>
  );
};

export default TemplatePage;
