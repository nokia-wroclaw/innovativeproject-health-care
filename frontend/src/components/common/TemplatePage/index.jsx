import React from "react";
import Header from "../../Header";
import Footer from "../../Footer";
import LoginModal from "../LoginModal";
import "./style.css";

const TemplatePage = props => {
  return (
    <div className="template-page-container">
      <LoginModal />
      <Header />
      <div className="template-page-content">{props.children}</div>
      <Footer />
    </div>
  );
};

export default TemplatePage;
