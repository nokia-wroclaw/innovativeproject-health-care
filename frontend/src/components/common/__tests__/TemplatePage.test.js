import React from "react";
import { shallow } from "enzyme";
import TemplatePage from "../TemplatePage";

describe("TemplatePage", () => {
  it("should match snapshot", () => {
    expect(shallow(<TemplatePage />)).toMatchSnapshot();
  });
});
