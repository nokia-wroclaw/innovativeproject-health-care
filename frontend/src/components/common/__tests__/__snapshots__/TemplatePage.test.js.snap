// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplatePage should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TemplatePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(LoginModal) />,
        <Connect(Header) />,
        <div
          className="template-page-content"
        />,
        <Footer />,
      ],
      "className": "template-page-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "template-page-content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(LoginModal) />,
          <Connect(Header) />,
          <div
            className="template-page-content"
          />,
          <Footer />,
        ],
        "className": "template-page-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "template-page-content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
